### PR TITLE
Converted two methods to use optionals

### PR DESCRIPTION
**Pull request by Adrijaned**
This is more of a proof of concept than a full fledged PR

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine/src/main/java/org/destinationsol/game/chunk/ChunkFiller.java [hunk 1]
```diff
@@ -176,19 +179,15 @@ private void fillEnemies(SolGame game, RemoveController removeController, ShipCo
         }
 
         for (int i = 0; i < enemyCount; i++) {
-            Vector2 enemyPosition = getFreeRndPos(game, chunkCenter);
-            FarShip ship = buildSpaceEnemy(game, enemyPosition, removeController, enemyConfig);
-            if (ship != null) {
+            Optional<Vector2> enemyPosition = getFreeRndPos(game, chunkCenter);
+            enemyPosition.ifPresent(position -> {
```
Comment (eviltak) [inline engine/src/main/java/org/destinationsol/game/chunk/ChunkFiller.java]: I'd prefer keeping the name of the introduced lambda parameter similar (or same if the compiler allows it) to the `Optional` variable name to retain context and clarity.

File: engine/src/main/java/org/destinationsol/game/chunk/ChunkFiller.java [hunk 2]
```diff
@@ -209,18 +208,17 @@ private void fillAsteroids(SolGame game, RemoveController remover, boolean forBe
...
-            float minSz = forBelt ? MIN_BELT_A_SZ : MIN_SYS_A_SZ;
-            float maxSz = forBelt ? MAX_BELT_A_SZ : MAX_SYS_A_SZ;
-            float sz = SolRandom.randomFloat(minSz, maxSz);
-            Vector2 speed = new Vector2();
-            SolMath.fromAl(speed, SolRandom.randomFloat(180), MAX_A_SPD);
-
-            FarAsteroid a = game.getAsteroidBuilder().buildNewFar(asteroidPos, speed, sz, remover);
-            game.getObjectManager().addFarObjNow(a);
+            Optional<Vector2> asteroidPos = getFreeRndPos(game, chunkCenter);
+            asteroidPos.ifPresent(position -> {
```
Comment (eviltak) [inline engine/src/main/java/org/destinationsol/game/chunk/ChunkFiller.java]: Same as above.

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by eviltak** [comment]
Looks good to me too.

**Comment by Cervator** [comment]
Tested and merged :-)

**Pull request merged**